config/zookeeper_persister.go

```diff
@@ -228,14 +230,24 @@ func (z *ZkConfigPersister) currentConfigEventListener() (<-chan zk.Event, error
 		}
 
 		configs[child] = data
+
+		// TODO(manik) replace this with a ZK node that tracks the latest version rather than deserializing each node
+		currentVersion := versionOf(data)
+
+		// TODO(manik) we can currently have multiple hashes with the same version; this needs to be fixed at the time
+		// of writing
+		if currentVersion >= latestHashVersion {
+			latestHashVersion = currentVersion
+			latestHash = child
+		}
 	}
 
 	if z.initialized {
 		logging.Printf("Re-establishing zookeeper watch on %v", z.path)
 	} else {
 		logging.Printf("Establishing zookeeper watch on %v", z.path)
 	}
-	config, _, ch, err := z.conn.GetW(z.path)
+	_, _, ch, err := z.conn.GetW(z.path)
```

Comment:
This probably warrants some kind of comment to say why you're ignoring the return value.

Reply:
that would help me understand what's going on as well

Reply:
Makes sense